### ISSUE TITLE
isssue in vignette

### ISSUE THREAD

**Issue by olivroy** [r-tmap/tmap#927]
This commit caused the following example to error 133f559c51c393186ef4a2afadc4c186793f8532.

What the correct code should be?

https://r-tmap.github.io/tmap/dev/articles/tmap_sneak_peek.html#aesthetics
```r
# tmap v4 (before the commit)
tm_shape(World) +
    tm_polygons(fill = "HPI", 
                fill.scale = tm_scale_intervals(values = "purple_green"), 
                fill.legend = tm_legend(title = "Happy Planet Index"))
```

![image](https://github.com/user-attachments/assets/919187fc-c5ef-4900-935e-ac1fecf957da)

**Comment by olivroy** [comment]
This also affects SDSR https://github.com/edzer/sdsr/actions/runs/10608550290

**Comment by olivroy** [comment]
Fixed by 9bcd02d18b9cdbf27b17fe89c5e58991255afae0
